internal/cli/atlas_alert_config_create.go

```diff
@@ -0,0 +1,234 @@
+
+	alertConfig := new(atlas.AlertConfiguration)
+
+	alertConfig.GroupID = opts.ProjectID()
+	alertConfig.EventTypeName = strings.ToUpper(opts.event)
+	alertConfig.Enabled = &opts.enabled
+
+	buildMatcher(opts, alertConfig)
+	buildMetricThreshold(opts, alertConfig)
+	buildNotification(opts, alertConfig)
+
+	return alertConfig
+}
+
+func buildNotification(opts *atlasAlertConfigCreateOpts, alertConfig *atlas.AlertConfiguration) {
+
+	notification := atlas.Notification{}
+	notification.TypeName = strings.ToUpper(opts.notificationType)
+	notification.DelayMin = &opts.notificationDelayMin
+	notification.IntervalMin = opts.notificationIntervalMin
+	notification.TeamID = opts.notificationTeamID
+	notification.Username = opts.notificationUsername
+
+	switch notification.TypeName {
+
+	case victor:
+		notification.VictorOpsAPIKey = opts.apiKey
+		notification.VictorOpsRoutingKey = opts.notificationVictorOpsRoutingKey
+
+	case slack:
+		notification.VictorOpsAPIKey = opts.apiKey
+		notification.VictorOpsRoutingKey = opts.notificationVictorOpsRoutingKey
+		notification.APIToken = opts.notificationToken
+
+	case datadog:
+		notification.DatadogAPIKey = opts.apiKey
+		notification.DatadogRegion = strings.ToUpper(opts.notificationRegion)
+
+	case email:
+		notification.EmailAddress = opts.notificationEmailAddress
+
+	case flowdock:
+		notification.FlowdockAPIToken = opts.notificationToken
+		notification.FlowName = opts.notificationFlowName
+		notification.OrgName = opts.notificationOrgName
+
+	case sms:
+		notification.MobileNumber = opts.notificationMobileNumber
+
+	case group, user, org:
+		notification.SMSEnabled = &opts.notificationSmsEnabled
+		notification.EmailEnabled = &opts.notificationEmailEnabled
+
+	case ops:
+		notification.OpsGenieAPIKey = opts.apiKey
+		notification.OpsGenieRegion = opts.notificationRegion
+
+	case pager:
+		notification.ServiceKey = opts.notificationServiceKey
+
+	}
+
+	alertConfig.Notifications = []atlas.Notification{notification}
+}
+
+func buildMetricThreshold(opts *atlasAlertConfigCreateOpts, alertConfig *atlas.AlertConfiguration) {
+	if opts.metricThresholdMetricName != "" {
+		metric := new(atlas.MetricThreshold)
+		metric.MetricName = strings.ToUpper(opts.metricThresholdMetricName)
+		metric.Operator = strings.ToUpper(opts.metricThresholdOperator)
+		metric.Threshold = opts.metricThresholdThreshold
+		metric.Units = strings.ToUpper(opts.metricThresholdUnits)
+		metric.Mode = strings.ToUpper(opts.metricThresholdMode)
+		alertConfig.MetricThreshold = metric
+	}
+}
+
+func buildMatcher(opts *atlasAlertConfigCreateOpts, alertConfig *atlas.AlertConfiguration) {
+	if opts.matcherFieldName != "" {
+		match := new(atlas.Matcher)
+		match.FieldName = strings.ToUpper(opts.matcherFieldName)
+		match.Operator = strings.ToUpper(opts.matcherOperator)
+		match.Value = strings.ToUpper(opts.matcherValue)
+		alertConfig.Matchers = []atlas.Matcher{*match}
+	}
+}
```

Comment:
there are multiple approaches for this, I don't think one is particularly better than the other
If you want to keep the approach where you modify an existing struct please reduce the area of your functions, don't pass the whole `atlas.AlertConfiguration`, pass the appropriate struct for the method name, be it a  `atlas.MetricThreshold` or `atlas.Matcher`, also please name this variable `out` to indicate to users of your function that that struct is being modified

Another approach you could have done:
```go
func (opts *atlasAlertConfigCreateOpts) newMatcher() *atlas.Matcher {
  return &atlas.Matcher{
    FieldName: strings.ToUpper(opts.matcherFieldName),
    Operator: strings.ToUpper(opts.matcherOperator),
    Value: strings.ToUpper(opts.matcherValue),
  }
}
```
And build the array in `buildAlertConfiguration` after doing the check for `opts.matcherFieldName != ""` in that method, same with `buildMetricThreshold`, and `buildNotification`, also the use of `new*` for methods that create a new reference to a struct is more common on go than `build*`